File: juego.go
//Title:Blackjack-Go
//Author:@RokCK
//Version:0.1.2

package main
import (
  "fmt"
  "math/rand"
  "time"
)

func DrawCard() string {
	suits := []string{"Treboles", "Diamantes", "Corazones", "Espadas"}
	ranks := []string{"2", "2", "2", "2", "3", "3", "3", "3", "4", "4", "4", "4", "5", "5", "5", "5", "6", "6", "6", "6", "7", "7", "7", "7", "8", "8", "8", "8", "9", "9", "9", "9", "10", "10", "10", "10", "Jota", "Jota", "Jota", "Jota", "Reina", "Reina", "Reina", "Reina", "Rey", "Rey", "Rey", "Rey", "As", "As", "As", "As"}
	rand.Seed(time.Now().UnixNano())
	suit := suits[rand.Intn(len(suits))]
	rank := ranks[rand.Intn(len(ranks))]
	return fmt.Sprintf("%s de %s", rank, suit)
}

func CardValue(card string) int {
	switch {
	case strings.HasPrefix(card, "As"):
		return 11
	case strings.HasPrefix(card, "Jota"),
		strings.HasPrefix(card, "Reina"),
		strings.HasPrefix(card, "Rey"):
		return 10
	default:
		return convertRankToValue(card)
	}
}

func convertRankToValue(card string) int {
	cardParts := strings.Split(card, " ")
	rank := cardParts[0]
	switch rank {
	case "2", "3", "4", "5", "6", "7", "8", "9", "10":
		value, _ := strconv.Atoi(rank)
		return value
	default:
		return 0
	}
}

func HandValue(hand []string) int {
	sum := 0
	aces := 0
	for _, card := range hand {
		value := CardValue(card)
		sum += value
		if value == 11 {
			aces++
		}
	}
	for sum > 21 && aces > 0 {
		sum -= 10
		aces--
	}
	return sum
}

func main() {
  fmt.Println("Hola Mundo")
}

File: juego.py
#Title:Blackjack-Python
#Author:@RokCK
#Version:0.1.2

import random

#Funciones
def draw_card():
    suits = ["Treboles", "Diamantes", "Corazones", "Espadas"]
    ranks = ["2", "2", "2", "2", "3", "3", "3", "3", "4", "4", "4", "4", "5", "5", "5", "5", "6", "6", "6", "6", "7", "7", "7", "7", "8", "8", "8", "8", "9", "9", "9", "9", "10", "10", "10", "10", "Jota", "Jota", "Jota", "Jota", "Reina", "Reina", "Reina", "Reina", "Rey", "Rey", "Rey", "Rey", "Rey", "As", "As", "As", "As"]
    return f"{random.choice(ranks)} of {random.choice(suits)}"

def card_value(card):
    rank = card.split()[0]
    if rank in ["Jota", "Reina", "Rey"]:
        return 10
    elif rank == "As":
        return 11
    else:
        return int(rank)

def hand_value(hand):
    value = sum(card_value(card) for card in hand)
    if value > 21 and "As" in [card.split()[0] for card in hand]:
        value -= 10
    return value

print("..:: BLACKJACK ::..")
hand = [draw_card(), draw_card()]
print(f"Tu mano: {hand[0]}, {hand[1]}")
hand_value = hand_value(hand)

if hand_value == 21:
    print("Felicidades. Ganaste!")
    quit()

while True:
    action = input("Decide: pides o te quedas? ").lower()

    if action == "p":
        hand.append(draw_card())
        print(f"Sacaste: {hand[-1]}")
        hand_value = hand_value(hand)

        if hand_value > 21:
            print("Te pasaste. Has perdido.")
            quit()
        elif hand_value == 21:
            print("Tu mano vale 21! Felicidades. Ganaste.")
            quit()
        else:
            print(f"Tu mano vale: {hand_value}")
    elif action == "q":
        print("Te quedas con una mano de {hand_value}.")
        break
    else:
        print("Opcion inexistente. Escribe 'p' o 'q'.")
